tests/test_python.py

```
@@ -13,21 +13,35 @@
+        pytest.fail(f"Compatible Python failed check: {e}")
+
+
+def test_find_compatible_pythons():
+    """Test finding compatible Python executables."""
+    pythons = python.find_compatible_pythons()
+
+    # Should find at least the current Python
+    assert len(pythons) >= 1
+
+    # All found pythons should be valid executables
+    import os
```

Ruff (0.12.2)

Redefinition of unused `os` from line 15

Remove definition: `os`

(F811)